dist/misc/uninstall-sgx-powhsm

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+# The name of the service unit that will be uninstalled
+SERVICE_NAME=powhsmsgx.service
+# The internal network name for the powHSM service
+NETWORK=powhsmsgx_net
+# The user name created during the installation
+USER=powhsm
+
+function error() {
+    if [ $? -ne 0 ]; then
+        echo "Error: $1"
+        exit 1
+    fi
+}
+
+function promptUserConfirmation() {
+    echo "Are you sure you want to uninstall the powHSM SGX service from this machine?"
+    read -p "Proceed? [y/N]: " confirm
+    if [ "$confirm" != "Y" ] && [ "$confirm" != "y" ]; then
+        echo "Uninstall cancelled."
+        exit 1
+    fi
+}
+
+function backupInstallDirectory() {
+    if [ ! -f "/etc/systemd/system/$SERVICE_NAME" ]; then
+        echo "Service unit file not found, skipping backup."
+        return
+    fi
+
+    WORKING_DIR=$(grep "^WorkingDirectory=" "/etc/systemd/system/$SERVICE_NAME" | cut -d'=' -f2)
+    if [ -z "$WORKING_DIR" ]; then
+        echo "Could not determine working directory from service unit. Aborting."
+        exit 1
+    fi
+
+    if [ ! -d "$WORKING_DIR" ]; then
+        echo "Working directory $WORKING_DIR does not exist, skipping backup."
+        return
+    fi
+
+    if [ -f "backup.tgz" ]; then
```

Comment:
Can we check this at the start, before actually doing anything? Otherwise e.g. the service would be disabled but then uninstallation wouldn't proceed.